frontend/src/components/file-explorer/CodeEditor.tsx

```diff
@@ -84,12 +96,19 @@ function CodeEditor(): JSX.Element {
     setSaveStatus("saving");
 
     try {
-      await saveFile(activeFilepath, code);
+      const newContent = fileState?.unsavedContent;
+      if (newContent) {
+        await saveFile(activeFilepath, newContent);
+      }
       setSaveStatus("saved");
       setShowSaveNotification(true);
-      setLastSavedContent(code);
-      setHasUnsavedChanges(false);
       setTimeout(() => setShowSaveNotification(false), 2000);
+      const newFileState = {
+        path: activeFilepath,
+        savedContent: newContent,
+        unsavedContent: newContent,
+      };
+      setTimeout(() => dispatch(addOrUpdateFileState(newFileState)), 2000);
```

Comment:
We should consider moving this into a `useEffect` that triggers when `saveStatus` changes. That way, we can also safely clear the timeout for the cases the user navigates away from the code editor.

What do you think?

Reply:
Good idea - gonna implement that now.